### PR TITLE
feat: pushdown aggr, limit and sort plan

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Push down `Sort`, `Limit` and `Aggr` plan under some conditions. 

As most intermediate stages are computed in datanode, the logical plan in frontend is very simple. I'm finding a way to make it more detailed

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by evenyag — APPROVED** [review]

File: src/query/src/dist_plan/commutativity.rs [hunk 1]
```diff
@@ -67,7 +82,17 @@ impl Categorizer {
             LogicalPlan::EmptyRelation(_) => Commutativity::NonCommutative,
             LogicalPlan::Subquery(_) => Commutativity::Unimplemented,
             LogicalPlan::SubqueryAlias(_) => Commutativity::Unimplemented,
-            LogicalPlan::Limit(_) => Commutativity::PartialCommutative,
+            LogicalPlan::Limit(limit) => {
+                // Only execute `fetch` on remote nodes.
+                // wait for https://github.com/apache/arrow-datafusion/pull/7669
+                if partition_cols.is_empty() && limit.fetch.is_some() {
```
Comment (killme2008) [inline src/query/src/dist_plan/commutativity.rs:88]: Is it possible for the `Limit` node to be commutative if we only fetch data from one datanode based on the query filter and partition rules?

Comment (waynexia) [reply src/query/src/dist_plan/commutativity.rs:88]: `partition_cols.is_empty()` is one of those cases. Infer from filter requires the filter to be strictly a superset of the partition rule, which is very limited and narrow. I don't give it a high priority to implement.

Comment (killme2008) [reply src/query/src/dist_plan/commutativity.rs:88]: >  Infer from filter requires the filter to be strictly a superset of the partition rule, which is very limited and narrow.

I don't think so, query data by partition columns is a common case. The user chooses the partition columns carefully.

Comment (waynexia) [reply src/query/src/dist_plan/commutativity.rs:88]: The partition values also have to match.

And this rule only cuts a few RPCs that won't return any rows. I don't think it's useless but there are many other rules that are much more useful.

Comment (killme2008) [reply src/query/src/dist_plan/commutativity.rs:88]: Ok, let's improve it step by step.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**